server/services/applicationService.test.ts

```diff
@@ -0,0 +1,67 @@
+import type { Request } from 'express'
+import { createMock } from '@golevelup/ts-jest'
+
+import UnknownPageError from './applicationForm/errors/unknownPageError'
+import ApplicationService from './applicationService'
+
+import pages from './applicationForm/pages'
+
+const FirstPage = jest.fn()
+const SecondPage = jest.fn()
+
+jest.mock('./applicationForm/pages', () => {
+  return {
+    'my-task': {},
+  }
+})
+
+pages['my-task'] = {
+  first: FirstPage,
+  second: SecondPage,
+}
+
+describe('ApplicationService', () => {
+  const service = new ApplicationService()
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+  describe('getCurrentPage', () => {
+    const request = createMock<Request>({ params: { task: 'my-task' } })
```

Comment:
Maybe a little low level for a draft PR but could we put this in a `beforeEach`?

Reply:
Something a little like this?
```suggestion
     describe('getCurrentPage', () => {
       let request;
       beforeEach(() => {
            request = createMock<Request>({ params: { task: 'my-task' } })
       })
      it('should return a page from a page list', () => {
        request.params.page = 'second'
   .....etc
```
something like that... So each time we use the request object it has been reset. My concern is that we're mutating the same request object that is used for each of the tests in this describe block. It works fine now but could make later changes painful.